File: 53/soln53.py
def nCr(n, r):
    def go(x, y, acc):
        if y == 0 or x == y:
            return acc
        return go(x - 1, y - 1, (x / y) * acc)
    return go(n, r, 1)

def main():
    numGreater = 0
    for i in range(23, 101):
        for j in range(i):
            if nCr(i, j) > 1000000:
                numGreater += 1 
    print(numGreater)

if __name__ == '__main__':
    main()

File: 53/soln53.hs
import System.IO

nCr :: Int -> Int -> Integer
nCr n r = round $ nCr' n r 1.0 where
    nCr' x y acc
        | y == 0 = acc
        | x == y = acc
        | otherwise = nCr' (x - 1) (y - 1) $!
            ((fromIntegral x / fromIntegral y) * acc)

main = do
    let allCombos = [nCr n r | n <- [23..100], r <- [10..100]]
        numQualified = length $ filter (\x -> x <= 1000000) allCombos
    print numQualified